lib/dependencies/ImportDependency.js

```diff
@@ -2,32 +2,49 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var ModuleDependency = require("./ModuleDependency");
-var DepBlockHelpers = require("./DepBlockHelpers");
-var WebpackMissingModule = require("./WebpackMissingModule");
+"use strict";
+const ModuleDependency = require("./ModuleDependency");
+const DepBlockHelpers = require("./DepBlockHelpers");
+const webpackMissingPromiseModule = require("./WebpackMissingModule").promise;
 
-function ImportDependency(request, block) {
-	ModuleDependency.call(this, request);
-	this.block = block;
+class ImportDependency extends ModuleDependency {
+	constructor(request, block) {
+		super(request);
+		this.block = block;
+	}
+
+	get type() {
+		return "import()";
+	}
 }
-module.exports = ImportDependency;
 
-ImportDependency.prototype = Object.create(ModuleDependency.prototype);
-ImportDependency.prototype.constructor = ImportDependency;
-ImportDependency.prototype.type = "import()";
-
-ImportDependency.Template = function ImportDependencyTemplate() {};
-
-ImportDependency.Template.prototype.apply = function(dep, source, outputOptions, requestShortener) {
-	var depBlock = dep.block;
-	var promise = DepBlockHelpers.getDepBlockPromise(depBlock, outputOptions, requestShortener, "import()");
-	var comment = "";
-	if(outputOptions.pathinfo) comment = "/*! " + requestShortener.shorten(dep.request) + " */ ";
-	if(promise && dep.module) {
-		source.replace(depBlock.range[0], depBlock.range[1] - 1, promise + ".then(__webpack_require__.bind(null, " + comment + JSON.stringify(dep.module.id) + "))");
-	} else if(dep.module) {
-		source.replace(depBlock.range[0], depBlock.range[1] - 1, "Promise.resolve(__webpack_require__(" + comment + JSON.stringify(dep.module.id) + "))");
-	} else {
-		source.replace(depBlock.range[0], depBlock.range[1] - 1, WebpackMissingModule.promise(dep.request));
+ImportDependency.Template = class ImportDependencyTemplate {
+	apply(dep, source, outputOptions, requestShortener) {
+		const depBlock = dep.block;
+		const promise = DepBlockHelpers.getDepBlockPromise(depBlock, outputOptions, requestShortener, "import()");
+		const comment = this.getOptionalComment(outputOptions.pathinfo, requestShortener.shorten(dep.request));
+
+		const content = this.getContent(promise, dep, comment);
+		source.replace(depBlock.range[0], depBlock.range[1] - 1, content);
+	}
+
+	getOptionalComment(pathinfo, shortenedRequest) {
+		if(!pathinfo) {
+			return "";
+		}
+
+		return `/*! ${shortenedRequest} */ `;
+	}
+
+	getContent(promise, dep, comment) {
+		if(dep.module) {
+			const prefix = promise ? promise : "Promise.resolve()";
```

Comment:
This increases bundle size compared to the original version. Please stay close to the original version when refactoring to ES6 and don't introduce functional changes.